app/src/main/java/com/keylesspalace/tusky/components/trending/TrendingFragment.kt

```diff
@@ -0,0 +1,306 @@
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_trending, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        setupSwipeRefreshLayout()
+        setupRecyclerView()
+
+        adapter.registerAdapterDataObserver(object : RecyclerView.AdapterDataObserver() {
+            override fun onItemRangeInserted(positionStart: Int, itemCount: Int) {
+                if (positionStart == 0 && adapter.itemCount != itemCount) {
+                    binding.recyclerView.post {
+                        if (getView() != null) {
+                            binding.recyclerView.scrollBy(
+                                0,
+                                Utils.dpToPx(requireContext(), -30)
+                            )
+                        }
+                    }
+                }
+            }
+        })
+
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewModel.uiState.collectLatest { trendingState ->
+                processViewState(trendingState)
+            }
+        }
+    }
+
+    private fun setupSwipeRefreshLayout() {
+        binding.swipeRefreshLayout.setOnRefreshListener(this)
+        binding.swipeRefreshLayout.setColorSchemeResources(R.color.tusky_blue)
+    }
+
+    private fun setupLayoutManager(columnCount: Int) {
+        binding.recyclerView.layoutManager = GridLayoutManager(context, columnCount).apply {
+            spanSizeLookup = object : SpanSizeLookup() {
+                override fun getSpanSize(position: Int): Int {
+                    return when (adapter.getItemViewType(position)) {
+                        TrendingPagingAdapter.VIEW_TYPE_HEADER -> columnCount
+                        TrendingPagingAdapter.VIEW_TYPE_TAG -> 1
+                        else -> -1
+                    }
+                }
+            }
+        }
+    }
+
+    private fun setupRecyclerView() {
+        val columnCount =
+            requireContext().resources.getInteger(R.integer.trending_column_count)
+        setupLayoutManager(columnCount)
+
+        binding.recyclerView.setHasFixedSize(true)
+        val divider = DividerItemDecoration(context, RecyclerView.VERTICAL)
+        binding.recyclerView.addItemDecoration(divider)
+
+        // CWs are expanded without animation, buttons animate itself, we don't need it basically
+        (binding.recyclerView.itemAnimator as SimpleItemAnimator).supportsChangeAnimations = false
+        binding.recyclerView.adapter = adapter
+    }
+
+    override fun onRefresh() {
+        viewModel.invalidate()
+    }
+
+    override fun onViewUrl(url: String) {
+        bottomSheetActivity.viewUrl(url, PostLookupFallbackBehavior.OPEN_IN_BROWSER)
+    }
+
+    override fun onViewTag(tag: String) {
+        startActivity(StatusListActivity.newHashtagIntent(requireContext(), tag))
```

Comment:
Open it with a slide-in animation please, it is also closed with one. (check e.g. [here](https://github.com/tuskyapp/Tusky/blob/a6b6a40ba6f940f8d612789c92524d5afeff078b/app/src/main/java/com/keylesspalace/tusky/components/search/fragments/SearchFragment.kt#L123) how to do it)

Reply:
Done :+1: